lib/ocpp/v201/connectivity_manager.cpp

```diff
@@ -121,6 +142,50 @@ bool ConnectivityManager::send_to_websocket(const std::string& message) {
     return this->websocket->send(message);
 }
 
+void ConnectivityManager::on_network_disconnected(const std::optional<int32_t> configuration_slot,
+                                                  const std::optional<OCPPInterfaceEnum> ocpp_interface) {
+
+    if (!configuration_slot.has_value() and !ocpp_interface.has_value()) {
+        EVLOG_warning << "Network disconnected. Not clear which network is disconnected: configuration slot and ocpp "
+                         "interface are empty";
+        return;
+    }
+
+    const int actual_configuration_slot = get_active_network_configuration_slot();
+    std::optional<NetworkConnectionProfile> network_connection_profile =
+        this->get_network_connection_profile(actual_configuration_slot);
+
+    if (!network_connection_profile.has_value()) {
+        EVLOG_warning << "Network disconnected. No network connection profile configured";
+    } else if ((configuration_slot.has_value() and (configuration_slot.value() == actual_configuration_slot)) or
+               (ocpp_interface.has_value() and
+                (ocpp_interface.value() == network_connection_profile.value().ocppInterface))) {
+        this->disconnect_websocket(ocpp::WebsocketCloseReason::GoingAway);
+        // Since there is no connection anymore: reconnect with the next available network connection profile.
+        reconnect(ocpp::WebsocketCloseReason::GoingAway);
+    }
+}
+
+bool ConnectivityManager::on_try_switch_network_connection_profile(const int32_t configuration_slot) {
+    if (!is_higher_priority_profile(configuration_slot)) {
+        return false;
+    }
+
+    EVLOG_info << "Trying to connect with higher priority network connection profile (configuration slots: "
+               << this->get_active_network_configuration_slot() << " --> " << configuration_slot << ").";
+
+    const std::optional<NetworkConnectionProfile> network_connection_profile_opt =
+        this->get_network_connection_profile(configuration_slot);
+    if (!network_connection_profile_opt.has_value()) {
+        EVLOG_warning << "Could not find network connection profile belonging to configuration slot "
+                      << configuration_slot;
+        return false;
+    }
+
+    reconnect(WebsocketCloseReason::Normal, get_configuration_slot_priority(configuration_slot));
```

Comment:
Don't we really need here to disconnect websocket first?

Reply:
You're right that it's not explicitly closed here. The websocket itself also does things based on the actions of this class, so I'll check if it's properly closed when you call this function.